### PR TITLE
feat(TRR): flag for different frame contents

### PR DESCRIPTION
**Pull request by ezavod**
TRR allows some data to be optional. The presence of box data, positions, velocities and forces are indicated by the flag.

### CODE REVIEW

File: include/xdrfile_trr.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ int read_trr_header(const char* fn, int* natoms, unsigned long* nframes, int64_t
 /* Read one frame of an open trr file. If either of x,v,f,box are
    NULL the arrays will be read from the file but not used.  */
 extern int read_trr(XDRFILE* xd, int natoms, int* step, float* t, float* lambda,
-                    matrix box, rvec* x, rvec* v, rvec* f);
+                    matrix box, rvec* x, rvec* v, rvec* f, uint8_t* has_prop);
```
Comment (Luthaf) [inline include/xdrfile_trr.h:54]: Is uint8_t guaranteed by C89 standard? If not, we might be safer of with `int`.

File: include/xdrfile_trr.h [hunk 2]
```diff
@@ -74,6 +74,12 @@ extern int write_trr(XDRFILE* xd, int natoms, int step, float t, float lambda,
  */
 #define TRR_MIN_HEADER_SIZE 72
 
+/* Flags to signal the existance of box, positions, velocities and forces in a frame */
+#define TRR_HASB 1 /* 0b0001 Box */
+#define TRR_HASX 2 /* 0b0010 Positions */
+#define TRR_HASV 4 /* 0b0100 Velocities */
+#define TRR_HASF 8 /* 0b1000 Forces */
```
Comment (Luthaf) [inline include/xdrfile_trr.h]: How about explicitly naming them `TRR_HAS_BOX`/`TRR_HAS_POSITIONS`/`TRR_HAS_VELOCITIES`/`TRR_HAS_FORCES`?

File: include/xdrfile_trr.h (hunk 1 above)
Comment (Luthaf) [reply include/xdrfile_trr.h:54]: Forget that, we are assuming availability of int64_t in xdr_seek/xdr_tell anyway. Maybe we can just use C99 everywhere ...

File: include/xdrfile_trr.h (hunk 2 above)
Comment (ezavod) [reply include/xdrfile_trr.h]: Makes sense and looks nicer.

**Pull request merged**